test/unit/matrix/test_retiled_matrix_ref.cpp

```diff
@@ -0,0 +1,276 @@
+//
+// Distributed Linear Algebra with Future (DLAF)
+//
+// Copyright (c) 2018-2024, ETH Zurich
+// All rights reserved.
+//
+// Please, refer to the LICENSE file in the root directory.
+// SPDX-License-Identifier: BSD-3-Clause
+//
+
+#include <iterator>
+#include <tuple>
+#include <utility>
+#include <vector>
+
+#include <dlaf/communication/communicator_grid.h>
+#include <dlaf/matrix/matrix.h>
+#include <dlaf/matrix/matrix_ref.h>
+
+#include <gtest/gtest.h>
+
+#include <dlaf_test/comm_grids/grids_6_ranks.h>
+#include <dlaf_test/matrix/util_matrix.h>
+#include <dlaf_test/matrix/util_matrix_senders.h>
+#include <dlaf_test/util_types.h>
+
+using namespace std::chrono_literals;
+
+using namespace dlaf;
+using namespace dlaf::matrix;
+using namespace dlaf::matrix::internal;
+using namespace dlaf::matrix::test;
+using namespace dlaf::comm;
+using namespace dlaf::test;
+using namespace testing;
+
+namespace ex = pika::execution::experimental;
+namespace tt = pika::this_thread::experimental;
+
+::testing::Environment* const comm_grids_env =
+    ::testing::AddGlobalTestEnvironment(new CommunicatorGrid6RanksEnvironment);
+
+template <typename Type>
+class RetiledMatrixRefLocalTest : public ::testing::Test {};
+
+TYPED_TEST_SUITE(RetiledMatrixRefLocalTest, MatrixElementTypes);
+
+template <typename Type>
+struct RetiledMatrixRefTest : public TestWithCommGrids {};
+
+TYPED_TEST_SUITE(RetiledMatrixRefTest, MatrixElementTypes);
+
+const std::vector<
+    std::tuple<LocalElementSize, TileElementSize, LocalTileSize, GlobalElementIndex, GlobalElementSize>>
+    local_sizes_tests({
+        // size, tile_size (target), tiles_per_block (target), distribution origin (ref), distribution size (ref)
+        {{8, 8}, {2, 2}, {2, 2}, {0, 0}, {4, 4}},
+        // {{8, 8}, {2, 2}, {2, 2}, {2, 2}, {4, 4}},
```

Comment:
To do?

Reply:
Yes, but I will remove it for this PR since it is not technically needed for the partial back-transformation and the PR is too big already.